### PR TITLE
fix: [CI-3395]: Fixed trigger audit trail

### PR DESCRIPTION
**Pull request by devkimittal**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31021)

### CODE REVIEW

File: 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import io.harness.annotations.dev.OwnedBy;
+import io.harness.outbox.OutboxEvent;
+import io.harness.outbox.api.OutboxEventHandler;
+
+import com.google.inject.Inject;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+
+@OwnedBy(PL)
```
Comment (harshjain12) [inline 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java:20]: It should be pipeline team

File: 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.ArgumentCaptor;
+
+public class PMSOutboxEventHandlerTest {
```
Comment (harshjain12) [inline 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java:47]: Add UT for trigger Update event

File: 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java (hunk 1 above)
Comment (devkimittal) [reply 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java:20]: My bad, I thought PL is pipeline.

**Comment by harshjain12** [comment]
Auditing update trigger is extremely imp. Please add UT for it.

File: 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        new TriggerDeleteEvent(accountIdentifier, orgIdentifier, projectIdentifier, triggerEntity);
+    String eventData = objectMapper.writeValueAsString(triggerDeleteEvent);
+    GlobalContext globalContext = new GlobalContext();
+    Principal principal =
+        new UserPrincipal(randomAlphabetic(10), randomAlphabetic(10), randomAlphabetic(10), randomAlphabetic(10));
+    SourcePrincipalContextData sourcePrincipalContextData =
+        SourcePrincipalContextData.builder().principal(principal).build();
+    globalContext.upsertGlobalContextRecord(sourcePrincipalContextData);
+
+    OutboxEvent outboxEvent = OutboxEvent.builder()
```
Comment (harshjain12) [inline 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java:118]: Use common method to generate event by passing event type.
Code is duplicated across UTs

File: 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+  @Test
+  @Category(UnitTests.class)
+  public void testTriggerDelete() throws Exception {
+    String accountIdentifier = randomAlphabetic(10);
+    String orgIdentifier = randomAlphabetic(10);
+    String projectIdentifier = randomAlphabetic(10);
+    String identifier = randomAlphabetic(10);
+    NGTriggerEntity triggerEntity =
+        NGTriggerEntity.builder().name(randomAlphabetic(10)).identifier(identifier).yaml(null).build();
```
Comment (harshjain12) [inline 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java:107]: why to pass null in yaml ?

File: 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    when(auditClientService.publishAudit(any(), any())).thenReturn(true);
+    assertEquals(true, pmsOutboxEventHandler.handle(outboxEvent));
+    verify(auditClientService, times(1)).publishAudit(auditEntryArgumentCaptor.capture(), any());
+    AuditEntry auditEntry = auditEntryArgumentCaptor.getValue();
+    assertEquals(Action.DELETE, auditEntry.getAction());
+  }
+
+  @Test
+  @Category(UnitTests.class)
+  public void testPipelineCreate() throws Exception {
```
Comment (harshjain12) [inline 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java:138]: remove duplicate code across UTs

File: 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java (hunk 4 above)
Comment (devkimittal) [reply 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java:107]: Not required in testing perspective

Comment (harshjain12) [reply 800-pipeline-service/src/test/java/io/harness/pms/outbox/PMSOutboxEventHandlerTest.java:107]: we need it.
We do convert yaml to object. What if conversion has some issue.

**Review by brijeshdhakar — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31021)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[800-pipeline-service/src/main/java/io/harness/PipelineServiceModule.java, line 407 at r1](https://reviewable.io/reviews/harness/harness-core/31021#-MwyvU0981Bbxt2642Uk:-MwyvU0981Bbxt2642Ul:b-sb1dtb) ([raw file](https://github.com/harness/harness-core/blob/477d3bac79f8a945bfd90df583d1fd489c59e82a/800-pipeline-service/src/main/java/io/harness/PipelineServiceModule.java#L407)):*
> ```Java
>         MapBinder.newMapBinder(binder(), String.class, OutboxEventHandler.class);
>     outboxEventHandlerMapBinder.addBinding(ResourceTypeConstants.TRIGGER).to(TriggerOutboxEventHandler.class);
>     outboxEventHandlerMapBinder.addBinding(ResourceTypeConstants.PIPELINE).to(PipelineOutboxEventHandler.class);
> ```

InputSet events will have resourceType=INPUT_SET, So we need to add a binding for inputSet events as well. And can you test the pipeline and inputSet audits flow?
___
*[800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java, line 33 at r1](https://reviewable.io/reviews/harness/harness-core/31021#-MwywXsq0o1bJgogDo65:-MwywXsq0o1bJgogDo66:bkeowdh) ([raw file](https://github.com/harness/harness-core/blob/477d3bac79f8a945bfd90df583d1fd489c59e82a/800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java#L33)):*
> ```Java
>   public boolean handle(OutboxEvent outboxEvent) {
>     try {
>       OutboxEventHandler handler = outboxEventHandlerMap.get(outboxEvent.getResource().getType());
> ```

Currently this would be null for inputSet and input set audits will be skiped. Please test pipeline and inputset audit flows are working properly after adding the inputset binding in pipelineServiceModule.

**Comment by devkimittal** [comment]
@brijeshdhakar Can you please check now?

File: 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  public boolean handle(OutboxEvent outboxEvent) {
+    try {
+      OutboxEventHandler handler = outboxEventHandlerMap.get(outboxEvent.getResource().getType());
+      if (handler != null) {
+        return handler.handle(outboxEvent);
+      }
+      return false;
+    } catch (Exception exception) {
+      log.error(
+          String.format("Unexpected error occurred during handling event of type %s", outboxEvent.getEventType()));
```
Comment (vistaarjuneja) [inline 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java:40]: Print the error as well

File: 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  @Override
+  public boolean handle(OutboxEvent outboxEvent) {
+    try {
+      OutboxEventHandler handler = outboxEventHandlerMap.get(outboxEvent.getResource().getType());
+      if (handler != null) {
+        return handler.handle(outboxEvent);
+      }
+      return false;
```
Comment (vistaarjuneja) [inline 800-pipeline-service/src/main/java/io/harness/pms/outbox/PMSOutboxEventHandler.java:37]: `return false` can be outside of both the try-catches

File: 800-pipeline-service/src/test/resources/trigger_new.yml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+pipeline:
+  identifier: secrethttp1
+  stages:
+    - stage:
+        identifier: qaStage
+        spec:
+          infrastructure:
+          infrastructureDefinition:
```
Comment (vistaarjuneja) [inline 800-pipeline-service/src/test/resources/trigger_new.yml:8]: Indentation required?

File: 800-pipeline-service/src/test/resources/trigger_old.yml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+pipeline:
+  identifier: secrethttp2
+  stages:
+    - stage:
+        identifier: qaStage
+        spec:
+          infrastructure:
+          infrastructureDefinition:
```
Comment (vistaarjuneja) [inline 800-pipeline-service/src/test/resources/trigger_old.yml:8]: Indentation required?

**Review by brijeshdhakar — COMMENTED** [review]
> Reviewed 1 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31021)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained.

File: 800-pipeline-service/src/test/resources/trigger_new.yml (hunk 8 above)
Comment (devkimittal) [reply 800-pipeline-service/src/test/resources/trigger_new.yml:8]: easy to look at

**Review by vistaarjuneja — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31021#-:-Mwza9iK08VMHYVkdNiz:bnfp4nl)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions.

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 4 of 6 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31021)* status: 1. unresolved discussions.

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 4 files at r1, 5 of 6 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31021#-:-MwzmooF8d1CT-82MmxE:bnfp4nl)* status: :shipit: complete! ✓

**Comment by diptiman-adak** [comment]
trigger ti0

**Comment by devkimittal** [comment]
trigger ti0

**Comment by devkimittal** [comment]
trigger ti0

**Comment by devkimittal** [comment]
trigger ti0

**Comment by harshjain12** [comment]
trigger messagecheck

**Pull request merged**